src/edu/stanford/dlss/wowza/SulWowza.java

```diff
@@ -12,76 +14,316 @@
+    String getDruid(String streamName)
+    {
+        String[] pieces = streamName.split("/");
+        String druid = pieces[0] + pieces[1] + pieces[2] + pieces[3];
+        if (DRUID_REGEX_PATTERN.matcher(druid).matches())
+            return druid;
+        else
+        {
+            getLogger().error(this.getClass().getSimpleName() + ": can't parse druid from streamName" +
+                    (streamName == null ? "" : ": " + streamName));
+            return null;
+        }
+    }
+
+    /** the media file to be streamed, as stored under the druid-tree;  needed to verify stacks token
+     * Assumption:  validateStreamName() was already called */
+    String getFilename(String streamName)
     {
-        getLogger().info("Tommy: SulWowza token = " + token);
-        return true;
+        String filename = streamName.substring(streamName.lastIndexOf('/') + 1);
+        if (filename.length() > 0)
+            return filename;
+        else
+        {
+            getLogger().error(this.getClass().getSimpleName() + ": can't parse fileName from streamName" +
+                    (streamName == null ? "" : ": " + streamName));
+            return null;
+        }
+    }
+
+    /** Assumption: stacksToken, druid, userIp and filename are all reasonable values (non-null, not empty, etc.) */
+    URL getVerifyStacksTokenUrl(String stacksToken, String druid, String filename, String userIp)
+    {
+        // TODO:  Url encode anything?   utf-8 charset affect anything? (filename, stacksToken)
+        String queryStr = "stacks_token=" + stacksToken + "&user_ip=" + userIp;
+        String fullUrl = stacksTokenVerificationBaseUrl + "/media/" + druid + "/" + filename + "/verify_token?" + queryStr;
+        try
+        {
+            return new URL(fullUrl);
+        }
+        catch (MalformedURLException err)
+        {
+            getLogger().error(this.getClass().getSimpleName() + " bad URL for stacks_token verification: ", err);
+            return null;
+        }
+    }
+
+    /** Assumption: verifyStacksTokenUrl is a valid URL */
+    boolean verifyTokenAgainstStacksService(URL verifyStacksTokenUrl)
+    {
+        try
+        {
+            HttpURLConnection stacksConn = (HttpURLConnection) verifyStacksTokenUrl.openConnection();
+            stacksConn.setRequestMethod("HEAD");
+            stacksConn.setConnectTimeout(stacksConnectionTimeout * 1000); // need milliseconds
+            stacksConn.setReadTimeout(stacksReadTimeout * 1000);  // need milliseconds
+            stacksConn.connect();
+            int status = stacksConn.getResponseCode();
+            getLogger().info(this.getClass().getSimpleName() + " sent verify_token request to " + verifyStacksTokenUrl);
+            getLogger().debug(this.getClass().getSimpleName() + " verify_token response code is " + String.valueOf(status));
+            if (status == HttpURLConnection.HTTP_OK)
+                return true;
+            else
+                return false;
+        }
```

Comment:
i think i'd be inclined to do a `GET` and check the text of the response, but maybe that's too OCD?  anyway, i'm totally 👌 with this.  and i guess this is the more RESTful approach anyway, to just use the status code?